LiteCore/Support/LogEncoder.hh

```diff
@@ -36,20 +37,22 @@ namespace litecore {
     public:
         static const uint8_t kMagicNumber[4];
 
-        LogEncoder(std::ostream &out);
+        LogEncoder(std::ostream &out, LogLevel level);
```

Comment:
I thought that was exactly the plan (Having a tool that interleaves all of the logs).  Isn't that what was indicated in the PRD review?  If it's unclear we should definitely confirm ASAP.

Reply:
When separate log files were first proposed (back in 1.x I think), I was all "WTF", but the rejoinder was that each file would include the higher levels, and this wouldn't be much overhead since there are pretty few Warning or Error logs. (But I guess with a Verbose log file, including all the Info logs _would_ be expensive.) I hadn't realized that this got taken out of the current proposal.

I worry that if the app crashes, the log levels might not be in sync. If the Info file got flushed more recently than the Warning file (because it's written more often), then afterwards you'd be missing some warnings that came before the last info messages...

Reply:
I don't think I have ever heard about the proposal you mentioned to include all of that in each successive level.  If the app crashes, then wouldn't the `atexit` handler flush all the logs?  Or does that not run for crashes?

Reply:
Yeah, you're right, I forgot about the atexit handler.